modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -3348,10 +3348,21 @@ def broadcast_apply_full_axis(
         PandasDataframe
             New Modin DataFrame.
         """
+
+        def get_partitions(df):
+            """Deal with the corner case if the "other" dataframe has no partitions."""
+            if df._partitions.size > 0:
+                return df._partitions
+            else:
+                empty_partition = self._partition_mgr_cls.create_partition_from_data(
+                    pandas.DataFrame(index=df.index, columns=df.columns)
+                )
```

Comment:
Maybe we should make this as a part of `combine` function?

Reply:
I think we can leave out create_partition_from_data. What do you think?

cc @arunjose696, @anmyachev

Reply:
Do you mean leave it as is? Then this will only work for `broadcast_apply_full ` function; if we want to use this approach with other functions, we will have to duplicate this logic.

Reply:
I think it would be better to leave the `create_partition_from_data` function as is. The issue is indeed  in `broadcast_apply_full_axis` as we dont send any data to remote function for empty partition dataframe, so I assume the issue would be better addressed in this function itself. 

The `combine` function is called only in merge()/join() so even if we use the logic there to create a empty partition(with index and columns), the `broadcast_apply_full_axis` function would fail for other operations eg `df.compare(df2)`, 

To use this approach in other functions, the alternative I think would be to make [get_partitions()](https://github.com/arunjose696/modin/blob/mergefix/modin/core/dataframe/pandas/dataframe/dataframe.py#L3352) a method of dataframe(and let it handle the case if partitions are empty) class would this be fine ?

Reply:
Done.